### PR TITLE
Sync with ESCOMP/CDEPS (2024-12) + Update global_control.nml.IN based on GW specs (#2528) + Sync with ESCOMP/CMEPS

### LINKED ISSUE #65: Sync with ESCOMP/main (2024-12-10)
Since February, a number of developments and bug fixes in authoritative ESCOMP upstream have been merged and should be brought to EMC fork, including
- data glacier model
- multilevel data ocean model
- add nuopc_shr_methods for UFS
- allow input of stream on unstructured mesh
- increase in filename length (baseline changing)
- extend JRA 1.5 streams
- change nextsw_cday calculation for CAM7 in cplhist mode
- add rpointer timestamp

### LINKED ISSUE #127: improve usability of dststatus fields
In CMEPS, there is currently functionality to return the dststatus field from the RH generation. This field provides [information ](https://earthsystemmodeling.org/docs/release/latest/ESMF_refdoc/node9.html#opt:regridstatus) about how each destination point was mapped. 

As currently implemented,  the file created by CMEPS when this feature is active requires additional manipulation by the user to visualize the fields. It would be more useful if the fields were written to a single file (vs one file per mapping per destination component) and on the destination component domain (vs the internal mesh representation).

### LINKED ISSUE #128: update NOAA-EMC from ESCOMP
A fix for Issue #127 has been merged at ESCOMP. That update needs to be brought to NOAA-EMC's fork. No answer changes are expected, but the update depends on updates in CDEPS for the nuopc_shr methods.

### PR DESCRIPTION
**Pull request by NickSzapiro-NOAA**
## Commit Queue Requirements:

- [x] Fill out all sections of this template.
- [x] All sub component pull requests have been reviewed by their code managers.
- [x] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [x] Commit 'test_changes.list' from previous step
---
## Description:

Since February, a number of developments and bug fixes in authoritative ESCOMP/CDEPS upstream have been merged and should be brought to EMC fork, including
- data glacier model
- multilevel data ocean model
- add nuopc_shr_methods for UFS
- allow input of stream on unstructured mesh
- increase in filename length (baseline changing)
- extend JRA 1.5 streams
- change nextsw_cday calculation for CAM7 in cplhist mode
- add rpointer timestamp

For CMEPS
- improved usability of the RegridStatus diagnostic returned during CMEPS RH formation
- use of share code under CDEPS to initialize Alarms

### Commit Message:

```
* UFSWM - Sync with ESCOMP/CDEPS (2024-12).  Sync with ESCOMP/CMEPS (2024-12). Make additional changes to global_control.nml.IN based on GW needs. 
  * CDEPS - Sync with ESCOMP/main (2024-12)
  * CMEPS - Sync with ESCOMP/main (2024-12)
```

### Priority:

* Normal

### UFSWM:

* Closes #2456 (Enabling RTs to run on CSPs Rocky8)
### Other:
* Closes NOAA-EMC/CDEPS#65
* Closes NOAA-EMC/CMEPS#128
* Closes NOAA-EMC/CMEPS#127 

### Sub component Pull Requests:

* CDEPS: NOAA-EMC/CDEPS#66
* CMEPS: NOAA-EMC/CMEPS#129
### UFSWM Blocking Dependencies:

* None

---
### Regression Test Changes (Please commit test_changes.list):

* PR Updates/Changes Baselines.

### Input data Changes:

* None.

### Library Changes/Upgrades:

* No Updates
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Comment by zach1221** [comment]
Hey, @NickSzapiro-NOAA . Would you mind if I added the CSP changes from PR[#2456](https://github.com/ufs-community/ufs-weather-model/pull/2456) to your PR here?

**Comment by NickSzapiro-NOAA** [comment]
Ok with me, @zach1221

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA I wonder if you'd consider re-opening this PR w/ the latest updates from CDEPS and then asking some priority on it? It has been languishing since mid-Oct and at this point I need the latest update from CDEPS in order to update CMEPS (I need the ``shr_get_rpointer_name`` in ``nuopc_shr_methods``). I have a feature branch I want to push to ESCOMP and we'll also have the flexible restart write feature to push up. If both of those need to wait for another CDEPS update cycle....

I've tested the latest CMEPS and it gives the same non-b4b result (in the hafs tests only, because of the filename length issue. (BTW, I think this should be marked ``baseline change`` and not ``baseline update``?)

**Comment by NickSzapiro-NOAA** [comment]
@DeniseWorthen Sure, I didn't know anyone was waiting on CDEPS. Sorry for the delay...looks like ~4-5 days per commit for the past few months

**Comment by DeniseWorthen** [comment]
No worries...the feature branch I want to push came out of the grid imprint work I've been doing, so it's only now that I realized I need the latest CDEPS. 

I could add a sync/ESCOMP PR for CMEPS also; that is what I tested and that will get us up-to-date w/ the upstreams for both CDEPS and CMEPS.

**Comment by NickSzapiro-NOAA** [comment]
@zach1221 Were these CSP changes all merged in https://github.com/ufs-community/ufs-weather-model/pull/2461 ?

**Comment by zach1221** [comment]
> @zach1221 Were these CSP changes all merged in #2461 ?

Yes, looks like everything merged with 2461.

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA Will we be able to add the CMEPS update to this PR?

**Comment by NickSzapiro-NOAA** [comment]
@DeniseWorthen This now gets up-to-date with ESCOMP/main. Pre-test should be ready by the morning

**Comment by NickSzapiro-NOAA** [comment]
@DeniseWorthen CDEPS part is ok now. I see your feature/updcmeps branch...happy to merge in here, if you'd like

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA Thanks. Let me just verify that I get the same file list change w/ the CMEPS update.

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA Sorry for the delay. I'm only running tests that use CMEPS on hercules to confirm that no other changes are expected but even so the reduced jobs are not yet done. So far no failures except hafs_regional_datm_cdeps_intel, which was expected from your PR.

**Comment by NickSzapiro-NOAA** [comment]
I'm ok with waiting for CDEPS but leave it up to @dpsarmie @FernandoAndrade-NOAA on whether to move ahead with #2528

**Comment by DeniseWorthen** [comment]
If there is a no-baseline change PR ready, then that is fine w/ me. I appreciate the effort move this PR up and get the CMEPS committed at the same time but it will be tomorrow before I'll have the results w/ the CMEPS PR included.

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA I'm still in the Q on hercules (??), so I ran Orion w/ your PR + my CMEPS update and got the same non-matching tests. Please point to my CMEPS branch in this PR. Thanks.

```
hafs_regional_docn intel
hafs_regional_docn_oisst intel
hafs_regional_datm_cdeps intel
```

My ufs branch feature/updcmeps is up-to-date and contains the required CMakeList change.

**Comment by NickSzapiro-NOAA** [comment]
@dpsarmie @DeniseWorthen Please let me know if I've brought in your changes correctly

File: CMEPS-interface/CMakeLists.txt
Comment (DeniseWorthen) [inline CMEPS-interface/CMakeLists.txt:49]: There was also a file removed (med_time_mod) in the CMakeLists

**Comment by dpsarmie** [comment]
#2528 changes seem to all be there. Thanks Nick

**Comment by DeniseWorthen** [comment]
@NickSzapiro-NOAA Looks good now. I also update the PR info for the CMEPS issue and PR.

**Comment by NickSzapiro-NOAA** [comment]
@FernandoAndrade-NOAA @jkbk2004 Should be ready now

**Review by dpsarmie — DISMISSED** [review]

**Review by DeniseWorthen — DISMISSED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
Ok, testing should be all set. I'll let subcomponent PRs know to continue with merging.

**Comment by NickSzapiro-NOAA** [comment]
Subcomponents were merged and updated here. Should be okay...please let me know

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by jkbk2004 — APPROVED** [review]

**Pull request merged**
